source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -349,6 +353,14 @@ def get_best_fallback_durations(self, main_mode):
             # So when
             # 1. overriding_fbs is empty, we fill in the result with main_fb's content
             # 2. uri can't be found in any of overriding_fbs, we fill in the result with main_fb's content
-            if all(duration <= fb.get(uri).duration for fb in overriding_fbs if fb.get(uri)):
-                res[uri] = duration
+            mode_durations = [(main_mode, duration)] + [
+                (mode, fb.get(uri).duration) for mode, fb in overriding_fbs if fb.get(uri)
+            ]
+            best_mode, best_duration = min(mode_durations, key=lambda o: o[1])
+            if best_mode != main_mode:
+                self._overrided_uri_map[main_mode][uri] = best_mode
+            res[uri] = best_duration
```

Comment:
Be careful for the case where `main_mode` and another `mode` have exactly the same duration, which one will be kept ? 
I would do something like : "I keep the first mode in the list overriding_modes that have a duration strictly lower than main_mode_duration (if any)"

Reply:
why would you keep the "First" strictly smaller one? why not the "Smallest" one?

Reply:
(Just for record)
https://docs.python.org/3/library/functions.html#min : If multiple items are minimal, the function returns the first one encountered.

so there won't be any ambiguities when several modes have the same samllest value.